scipy/stats/_relative_risk.py

```diff
@@ -0,0 +1,197 @@
+
+from collections import namedtuple
+import numpy as np
+from scipy.special import ndtri
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+
+class RelativeRiskResult:
+
+    def __init__(self, exposed_cases, exposed_total,
+                 control_cases, control_total,
+                 relative_risk):
+        self.relative_risk = relative_risk
+        self.exposed_cases = exposed_cases
+        self.exposed_total = exposed_total
+        self.control_cases = control_cases
+        self.control_total = control_total
+
+    def __repr__(self):
+        return f"RelativeRiskResult(relative_risk={self.relative_risk})"
+
+    def confidence_interval(self, confidence_level=0.95):
+        """
+        Compute the confidence interval for the relative risk.
+
+        The confidence interal is computed using the Katz method
+        ([1]_, [2]_).
+
+        Parameters
+        ----------
+        confidence_level : float, optional
+            The confidence level to use for the confidence interval.
+            Default is 0.95.
+
+        Returns
+        -------
+        ci : namedtuple
+            The return value is a namedtuple with named fields
+            ``low`` and ``high``.
+
+        References
+        ----------
+        .. [1] Hardeo Sahai and Anwer Khurshid, Statistics in Epidemiology,
+               CRC Press LLC, Boca Raton, FL, USA (1996).
+        .. [2] D. Katz, J. Baptista, S. P. Azen and M. C. Pike, "Obtaining
+               confidence intervals for the risk ratio in cohort studies",
+               Biometrics, 34, 469-474 (1978).
+
+        Examples
+        --------
+        >>> from scipy.stats import relative_risk
+        >>> result = relative_risk(exposed_cases=10, exposed_total=75,
+        ...                        control_cases=12, control_total=225)
+        >>> result.relative_risk
+        2.5
+        >>> result.confidence_interval()
+        ConfidenceInterval(low=1.1261564003469628, high=5.549850800541033)
+        """
+        if not 0 <= confidence_level <= 1:
+            raise ValueError('confidence_level must be in the interval '
+                             '[0, 1].')
+
+        alpha = 1 - confidence_level
+        z = ndtri(1 - alpha/2)
+        rr = self.relative_risk
```

Comment:
this one should maybe also have a check on non-negativity (wherever other checks wind up)-seeing after looking at the calculation a second time-right?

Reply:
`self.relative_risk` is computed by the main function.  If I ensure that the inputs are nonnegative then `self.relative_risk` will not be negative.  Is that what you are concerned about?

It is possible for `self.relative_risk` to be 0, inf or nan, and the code that computes the confidence interval should handle these cases "nicely".  I'll figure out what "nicely" means as I update the code to be more careful about the edge cases. :)

Reply:
Done.  I used the same convention as used by `riskratio` in the R library `epitools`.